packages/search/src/Search.ts

```diff
@@ -133,6 +177,58 @@ export class Search extends Textfield {
         this.inputElement.setAttribute('type', 'search');
     }
 
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        this._firstUpdateAfterConnected = true;
+        this.requestUpdate();
+    }
+
+    public override disconnectedCallback(): void {
+        // Cleanup all form & button event listeners and remove form element from DOM
+        this.formAbortController?.abort();
+        this.clearButtonAbortController?.abort();
+        this.clearButtonAbortController = undefined;
+        this.form.remove();
+
+        super.disconnectedCallback();
+    }
+
+    protected override updated(changedProperties: PropertyValues): void {
+        super.updated(changedProperties);
+        // Adding this here instead of firstUpdated because we want to make sure
+        // this is called again on the first update after a previous disconnect
+        if (this._firstUpdateAfterConnected) {
+            this._firstUpdateAfterConnected = false;
+            this.firstUpdateAfterConnected();
+        }
+
+        if (changedProperties.has('value')) {
+            this._manageClearButtonListeners();
+        }
+    }
+
+    protected override firstUpdateAfterConnected(): void {
+        super.firstUpdateAfterConnected();
+
+        this.formAbortController = new AbortController();
```

Comment:
Instead of creating a new AbortController instance every time the method is called, you can initialize it once in the constructor or connectedCallback and reuse it

Reply:
I can't do that because once the `controller.abort()` method has been called its `signal` can't be reused to add the event listeners again. So the current flow is as follows: 
- on first update after `connected` I create an instance of `AbortController`
- I use the controller's `signal` to add the form event listeners
- the form event listeners are removed on `disconnectedCallback` by calling `controller.abort()`